### PR TITLE
SvelteKit: Automatically support Kit-specific features

### PR DESCRIPTION
**Pull request by JReinhold**
This PR keeps the `vite-plugin-sveltekit-setup` that handles many Svelte**Kit**-specific features, so they get supported out of the box in Storybook.

Currently this PR is blocked by the issue that **built** Storybooks will always show the "No story" warning on initial load/reload. Navigating between stories afterwards works, and everything works fine in dev.
This issue is also causing the iframe viewmode to be broken.

Once that has been fixed, this should be good to merge.

To test this out:

1. `git checkout origin/sveltekit-keep-compile-plugin`
2. `yarn task --task sandbox --template svelte-kit/skeleton-js --start-from install`
3. `cd sandbox/svelte-kit-default-js`
4. `yarn build-storybook`
5. `npx http-serve storybook-static`

### CODE REVIEW

**Comment by ndelangen** [comment]
Is this part of the 7.0 beta project? What column does this belong in?

**Comment by ndelangen** [comment]
I had to allow for `vite` version 4, because that's what is used in sveltekit, it seems.

I also extracted that into it's own PR: https://github.com/storybookjs/storybook/pull/20294

**Comment by ndelangen** [comment]
@JReinhold I do not see the situation you describe happening.

Maybe there's a step missing in your steps to reproduce?

I ran your steps on this branch and got this static-storybook:
[storybook-static.zip](https://github.com/storybookjs/storybook/files/10240160/storybook-static.zip)

I do not see the "no story" UI show up"
Screenshot 2022-12-15 at 20 48 00

**Comment by ndelangen** [comment]
what is making unit tests fail?!?!

File: code/frameworks/sveltekit/src/preset.ts [hunk 1]
```diff
@@ -18,8 +18,7 @@ export const viteFinal: NonNullable<StorybookConfig['viteFinal']> = async (confi
   plugins = await withoutVitePlugins(plugins, [
     // pre @sveltejs/kit@1.0.0-next.574
```
Comment (benmccann) [inline code/frameworks/sveltekit/src/preset.ts:19]: I wouldn't worry supporting anything pre-1.0

Comment (JReinhold) [reply code/frameworks/sveltekit/src/preset.ts:19]: you're right, we should remove that now. I believe this was written before the 1.0 launch.

File: code/frameworks/sveltekit/README.md [hunk 2]
```diff
@@ -111,6 +112,16 @@ yarn remove @storybook/builder-vite
...
 
+### Error: `Cannot read properties of undefined (reading 'disable_scroll_handling')` in preview
+
+> Some stories don't load, instead they show the following error in the preview:
+>
+> ```
+> Cannot read properties of undefined (reading 'disable_scroll_handling')
+> ```
+
+You'll experience this if anything in your story is importing from `$app/forms` or `$app/navigaiton`, which is currently not supported. To get around this, separate your component into a shallow parent component that imports what's needed and passes it to a child component via props. This way you can write stories for your child component and mock any of the necessary modules by passing props in.
```
Comment (benmccann) [inline code/frameworks/sveltekit/README.md:123]:
```suggestion
You'll experience this if anything in your story is importing from `$app/forms` or `$app/navigation`, which is currently not supported. To get around this, separate your component into a shallow parent component that imports what's needed and passes it to a child component via props. This way you can write stories for your child component and mock any of the necessary modules by passing props in.
```

**Comment by JReinhold** [comment]
> I just tested and it worked fine for me as well. I did run into a separate issue: https://github.com/storybookjs/storybook/issues/20595. But I didn't see the "no story" UI, so I think this could be merged after being rebased

Alright, then let's clean it up and merge it, see if it was just my setup that was bad. Did you try having a story that actually imported and used all the supported SvelteKit modules? I wonder if that could cause it, as I had such a story. (I'll see if I can dig it out from somewhere)

> 
> > `npx http-serve storybook-static`
> 
> I wouldn't have known this was a thing you could do without reading the PR description. Should we add it to the `package.json`? Or the readme?

I think this is generally documented a few places in the Storybook docs, mainly here: https://storybook.js.org/docs/react/sharing/publish-storybook
I don't think we need to specifically point this out in relation to SvelteKit then? But the SvelteKit docs could do a better job of pointing the user to the Storybook docs though.

**Comment by benmccann** [comment]
> Did you try having a story that actually imported and used all the supported SvelteKit modules?

No, I didn't. It's possible that could be the cause

File: code/frameworks/sveltekit/src/preset.ts [hunk 3]
```diff
@@ -18,8 +18,7 @@ export const viteFinal: NonNullable<StorybookConfig['viteFinal']> = async (confi
...
     'vite-plugin-svelte-kit',
```
Comment (benmccann) [inline code/frameworks/sveltekit/src/preset.ts:20]:
```suggestion
```

**Comment by benmccann** [comment]
@JReinhold I just did `import { dev } from '$app/environment';` and included the value in `Button.svelte` and it worked fine. I don't know if there's another value you were importing that might have caused issues. This seems like it's probably in a good enough place that we can merge it as is and fix any bugs in followups

**Review by benmccann — APPROVED** [review]

**Pull request merged**